### ISSUE TITLE
Style/StringConcatenation and Rails HTML helpers

### ISSUE THREAD

**Issue by DawidJanczak** [rubocop/rubocop#8503]
The following piece of code `content_tag(:b, 'Bold') + ' Text'` gets flagged by the `Style/StringConcatenation`, but the result is different if we were to use interpolation. `content_tag` generates an instance of `ActiveSupport::SafeBuffer` that has `+` overridden and so:

```
t = content_tag(:b, 'Bold') + ' Text'
t.html_safe? # => true
```
but
```
t = "#{content_tag(:b, 'Bold')} Text"
t.html_safe? # => false
```

--------

## Expected behavior

`Style/StringConcatenation` shouldn't get triggered for instances of `ActiveSupport::SafeBuffer`.

## Actual behavior

`Style/StringConcatenation` gets triggered for instances of `ActiveSupport::SafeBuffer`.

## Steps to reproduce the problem

See above.

## RuboCop version

0.89.0

**Comment by jonas054** [comment]
Setting
```yaml
Style/StringConcatenation:
  Mode: conservative
```
skips any construct where `Style/StringConcatenation` can't determine the type of the left operand, i.e., where the left operand isn't a string literal. 

The default mode is `aggressive`, but when inspecting Rails code, especially in `views` or `helpers`, it seems more appropriate to use `conservative`.
